README/RCodingStyle.hxx

```diff
@@ -0,0 +1,171 @@
+   /// functions. See the implementation in the source file.
+   static const std::string &StaticFunc();
+
+public:
+   // THIRD section: methods, starting with constructors, then alphabetical order.
+
+   /// Use `= default` inside the class whenever possible
+   RExampleClass() = default;
+
+   /// Classes with virtual functions must have a virtual destructor. If all other functions are
+   /// pure virtual or inlined, the destructor should be outlined to pin the class's vtable to an
+   /// object file. I.e. say `= default` in the source, not here.
+   virtual ~RExampleClass();
+
+   /// Virtual functions are still fine (though we try to avoid them as they come with a call
+   /// and optimization penalty). See `RCodingStyle` for overriding virtual functions.
+   virtual void AVirtualFunction() = 0;
+
+   /// Static functions line up alphabetically with the others.
+   static SeqULL_t GetSeq() { return fkSeq; }
+
+   /// "Free" (i.e. non-class member) operators should be declared as "hidden friends".
+   /// This makes them available only when needed, instead of adding to the long list of
+   /// "no matching overload for a+b, did you mean ...(hundreds of unrelated types)..."
+   friend RExampleClass operator+(const RExampleClass &a, const RExampleClass &b) {
+      return {};
+   }
+
+};
+
+///\class RCodingStyle
+/// This is exhaustive documentation of the class template, explaining the constraints
+/// on the template parameters, e.g. `IDX` must be positive, `T` must be  copyable.
+/// Disabled if `T` is of reference type.
+
+template <int IDX, // Template parameters are all-capital letters
+          class T, // We use "class", not "typename"; `T` is fine for generic class names
+          // we use enable-if through unnamed template parameters and provide a `assert`-style message
+          // to make the diagnistics more understandable.
+          class = std::enable_if<"Must only be used with non-reference types"
+                                 && !std::is_reference_type<T>::value>
+         >
+class RCodingStyle: public RExampleClass {
+private:
+   T fMember; ///< The wrapped value.
+public:
+   // Use static_assert excessively: it makes diagnostics much more readable,
+   // and helps readers of your code.
+   static_assert(IDX > 0, "IDX must be >0.")
+
+   /// Defaulted default constructor.
+   RCodingStyle() = default;
+
+   // Use `= delete` instead of private, unimplemented.
+   // To avoid copy and move construction and assignment, `delete` the copy
+   // constructor and copy-assignment operator:
+   RCodingStyle(const RCodingStyle&) = delete;
+   RCodingStyle& operator=(const RCodingStyle&) = delete;
+
+
+   /// Virtual destructors of derived classes may be decorated with `virtual` (even though
+   /// they "inherit" the virtuality of the virtual base class destructor).
+   virtual ~RCodingStyle() {}
+
+   /// Overridden virtual functions do not specify `virtual` but `override` or `final`.
```

Comment:
I would say _either_ `override` or `final`: exactly one of the two, as per CppCoreGuidelines